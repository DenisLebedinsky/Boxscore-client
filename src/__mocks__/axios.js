export default {
  get: params => {
    return Promise.resolve({
      data: {
        league: "NBA",
        away_team: {
          abbreviation: "OKC",
          first_name: "Oklahoma City"
        },
        home_team: {
          abbreviation: "MIA",
          first_name: "Miami"
        },
        away_period_scores: [26, 23, 22, 35],
        home_period_scores: [31, 28, 36, 26],
        event_information: {
          status: "completed"
        },
        extentions: {
          away_totals: {
            points: 106
          },
          home_totals: {
            points: 121
          },
          home_batter_totals: {
            hits: 13,
            batting_highlights: "5-29, HR, 3 R, 3 RBI, 2B"
          },
          away_batter_totals: {
            hits: 7,
            batting_highlights: "10-33, 4 R, 4 RBI, 2B, SB"
          }
        }
      }
    });
  }
};
