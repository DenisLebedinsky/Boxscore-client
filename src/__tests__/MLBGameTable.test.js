import React from "react";
import ReactDOM from "react-dom";
import { mount } from "enzyme";
import "jest-enzyme";
import MLBGameTable from "../components/MLBGameTable";
import axios from "../__mocks__/axios";

jest.mock("axios");

describe("NBAGameTable", () => {
	const getSpy = jest.spyOn(axios, "get");


  const NBAGame = mount(<MLBGameTable />);

  it("render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<MLBGameTable />, div);
  });

  it("check contain block", () => {
    const boxscore = <div className="tableScore" />;
    expect(NBAGame).toContainReact(boxscore);
  });

  it("check call axios.get", () => {
    expect(getSpy).toBeCalled();
	});
	
  it("check contain child block", () => {
    expect(NBAGame).toMatchSnapshot();
	});
	
});
