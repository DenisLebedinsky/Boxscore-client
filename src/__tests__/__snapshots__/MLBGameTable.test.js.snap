// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NBAGameTable check contain child block 1`] = `
<Component>
  <MLBGameTable
    fetchData={[Function]}
    getScoreTableUI={[Function]}
    subscribeIo={[Function]}
  >
    <div>
      <div
        className="boxscore"
      >
        <div
          className="tableScore"
        />
      </div>
    </div>
  </MLBGameTable>
</Component>
`;
